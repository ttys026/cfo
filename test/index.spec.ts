// test/index.spec.ts
import { env, createExecutionContext, waitOnExecutionContext } from 'cloudflare:test';
import { describe, it, expect, vitest } from 'vitest';
import worker from '../src/index';

// For now, you'll need to do something like this to get a correctly-typed
// `Request` to pass to `worker.fetch()`.
const IncomingRequest = Request<unknown, IncomingRequestCfProperties>;

describe('should get item correctly', () => {
	it('https://ogp.me/', async () => {
		const request = new IncomingRequest('http://localhost/?url=https://ogp.me/');
		const ctx = createExecutionContext();
		const response = await worker.fetch(request, env, ctx);
		await waitOnExecutionContext(ctx);
		expect(await response.json()).toMatchSnapshot();
	});
	it('https://google.com/', async () => {
		const request = new IncomingRequest('http://localhost/?url=https://google.com/');
		const ctx = createExecutionContext();
		const response = await worker.fetch(request, env, ctx);
		await waitOnExecutionContext(ctx);
		expect(await response.json()).toMatchSnapshot();
	});
});

describe('should bring user-agent header', () => {
	it('https://codesandbox.io/p/sandbox/7t8chd', async () => {
		const request = new IncomingRequest('http://localhost/?url=https://codesandbox.io/p/sandbox/7t8chd');
		const ctx = createExecutionContext();
		const response = await worker.fetch(request, env, ctx);
		await waitOnExecutionContext(ctx);
		expect(await response.json()).toMatchSnapshot();
	});
});

describe('should return error', () => {
	it('https://some-url-that-does-not-work.abcde/', async () => {
		const request = new IncomingRequest('http://localhost/?url=https://some-url-that-does-not-work.abcde/');
		const ctx = createExecutionContext();
		const response = await worker.fetch(request, env, ctx);
		await waitOnExecutionContext(ctx);
		const json = await response.json();
		expect(json).toEqual({ error: 'Fetch failed' });
	});
});
