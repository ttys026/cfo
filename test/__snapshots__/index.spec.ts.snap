// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should bring user-agent header > https://codesandbox.io/p/sandbox/7t8chd 1`] = `
{
  "icon": "https://codesandbox.io/favicon.ico",
  "og:description": "promise  ",
  "og:image": "https://codesandbox.io/api/v1/sandboxes/7t8chd/screenshot.png",
  "og:image:alt": "A preview of promise",
  "og:image:height": "630",
  "og:image:width": "1200",
  "og:title": "promise - Codesandbox",
  "og:type": "article",
  "og:url": "https://codesandbox.io/s/bold-mcnulty-7t8chd",
}
`;

exports[`should get item correctly > https://google.com/ 1`] = `
{
  "icon": "https://google.com/favicon.ico",
  "og:description": "Google",
  "og:title": "Google",
}
`;

exports[`should get item correctly > https://ogp.me/ 1`] = `
{
  "icon": "https://ogp.me/favicon.ico",
  "og:description": "The Open Graph protocol enables any web page to become a rich object in a social graph.",
  "og:image": "https://ogp.me/logo.png",
  "og:image:alt": "The Open Graph logo",
  "og:image:height": "300",
  "og:image:type": "image/png",
  "og:image:width": "300",
  "og:title": "Open Graph protocol",
  "og:type": "website",
  "og:url": "https://ogp.me/",
}
`;
